protocol/communities/manager.go

```diff
@@ -1526,6 +1596,154 @@ func (m *Manager) IsSeedingHistoryArchiveTorrent(communityID types.HexBytes) boo
 	return ok && torrent.Seeding()
 }
 
+func (m *Manager) DownloadHistoryArchivesByMagnetlink(communityID types.HexBytes, magnetlink string) ([]string, error) {
+
+	id := communityID.String()
+	ml, _ := metainfo.ParseMagnetUri(magnetlink)
+
+	m.logger.Debug("Adding torrent via magnetlink for community", zap.Any("id", id), zap.Any("magnetlink", magnetlink))
+	torrent, err := m.torrentClient.AddMagnet(magnetlink)
+	if err != nil {
+		return nil, err
+	}
+	m.torrentTasks[id] = ml.InfoHash
+
+	<-torrent.GotInfo()
+	files := torrent.Files()
+
+	i, ok := findIndexFile(files)
+	if !ok {
+		// We're dealing with a malformed torrent, so don't do anything
+		return nil, nil
+	}
+
+	indexFile := files[i]
+
+	m.logger.Debug("Downloading history archive index")
+	indexFile.Download()
+	for {
+		if indexFile.BytesCompleted() == indexFile.Length() {
+			break
+		}
+	}
+
+	index, err := m.LoadHistoryArchiveIndexFromFile(communityID)
+	if err != nil {
+		return nil, err
+	}
+
+	var archiveIDs []string
+
+	archiveHashes := make(archiveMDSlice, 0, len(index.Archives))
+
+	for hash, metadata := range index.Archives {
+		archiveHashes = append(archiveHashes, &archiveMetadata{hash: hash, from: metadata.Metadata.From})
+	}
+
+	sort.Sort(archiveHashes)
```

Comment:
Here, we're creating the slice of the archive metadata to make it sortable and then sort it, making newer archives come first.